### ISSUE TITLE
🐛 [BUG] - unionml init fails on Python 3.11

### ISSUE THREAD

**Issue by DavidMertz** [unionai-oss/unionml#214]
### Description

In a Python 3.11 environment, `pip install unionml` succeeds, however:

```
(unionai) 2021-~ % unionml init practice
Traceback (most recent call last):
  File "/home/davidmertz/miniconda3/envs/unionai/bin/unionml", line 5, in <module>
    from unionml.cli import app
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/site-packages/unionml/__init__.py", line 4, in <module>
    from unionml.dataset import Dataset
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/site-packages/unionml/dataset.py", line 18, in <module>
    from flytekit.extras.sqlite3.task import SQLite3Task
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/site-packages/flytekit/__init__.py", line 165, in <module>
    from flytekit.core.base_sql_task import SQLTask
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/site-packages/flytekit/core/base_sql_task.py", line 4, in <module>
    from flytekit.core.base_task import PythonTask, TaskMetadata
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/site-packages/flytekit/core/base_task.py", line 26, in <module>
    from flytekit.configuration import SerializationSettings
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/site-packages/flytekit/configuration/__init__.py", line 473, in <module>
    @dataclass(init=True, repr=True, eq=True, frozen=True)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/dataclasses.py", line 1211, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/dataclasses.py", line 959, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/davidmertz/miniconda3/envs/unionai/lib/python3.11/dataclasses.py", line 816, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'flytekit.configuration.S3Config'> for field s3 is not allowed: use default_factory
```

### Reproduction URL

N/A

### Reproduction steps

```bash
1. conda activate <python=3.11 environment>
2. pip install unionml
3. unionml init practice
```

### Screenshots

```bash
![DESCRIPTION](LINK.png)
```

### OS

Linux

**Comment by DavidMertz** [comment]
To clarify, I wouldn't say lack of support for Python 3.11 is a *bug* in itself. But the fact the requirements for install allow installation on a version where the tool is non-functional is a bug.  Limiting the supported version range would fix the issue narrowly, but of course supporting 3.11 is better (but this probably depends on updated dependencies).

**Comment by cosmicBboy** [comment]
flytekit doesn't support 3.11 yet and as a result unionml doesn't either. Will need to update [this](https://github.com/unionai-oss/unionml/blob/main/setup.py#L36) to restrict the python version allowed for this package.

**Comment by thomasjpfan** [comment]
With https://github.com/flyteorg/flytekit/pull/1557 merged and shipped in 1.5.0, `unionml init practice` works on Python 3.11 and this issue can be closed.
